### PR TITLE
Add og tags and seo support

### PR DESCRIPTION
**Pull request by nzambello**
Fixes: #252 

Closes: #2141 
Closes: #2132 

Merged both the PRs into a single one.

### CODE REVIEW

File: src/components/theme/View/View.jsx [hunk 1]
```diff
@@ -220,8 +222,65 @@ class View extends Component {
...
+          <meta
+            property="og:url"
+            content={
+              this.props.content.seo_canonical_url ||
+              toPublicURL(this.props.content['@id'])
+            }
+          />
+          {this.props.content.image?.scales?.large?.download ||
+            (this.props.content.opengraph_image?.scales?.large?.download && (
+              <>
```
Comment (giuliaghisini) [inline src/components/theme/View/View.jsx:253]: this throws errors.. each <meta> it must not be enclosed in a fragment

**Review by giuliaghisini — CHANGES_REQUESTED** [review]

File: src/components/theme/View/View.jsx (hunk 1 above)
Comment (nzambello) [reply src/components/theme/View/View.jsx]: 👍

**Review by giuliaghisini — APPROVED** [review]

File: src/components/theme/View/View.test.jsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ beforeAll(() => {
       ECONNREFUSED: () => <div className="ECONNREFUSED" />,
     },
   });
+  config.settings.publicURL = 'https://plone.org';
```
Comment (tiberiuichim) [inline src/components/theme/View/View.test.jsx:23]: Can we have a safer default? Maybe take RAZZLE_API_PATH, or a variant, as the default?

Comment (nzambello) [reply src/components/theme/View/View.test.jsx:23]: This is only the test, the setting itself is in the config

Comment (sneridagh) [reply src/components/theme/View/View.test.jsx:23]: For the test is fine I guess. The config in the tests resets on every test.

**Review by sneridagh — COMMENTED** [review]
LGTM, check the single comment.

**Comment by tisto** [comment]
@jackahl can you please look into this PR as well and confirm that this is in sync with the settings that we have/use in our projects?

**Comment by jackahl** [comment]
> @jackahl can you please look into this PR as well and confirm that this is in sync with the settings that we have/use in our projects?

Checked. Is in sync with how we have it.

**Comment by sneridagh** [comment]
@nzambello @giuliaghisini there was an error in the code, could you take a look and confirm that a lead `image` wins over an `opengraph_image`?

**Review by sneridagh — APPROVED** [review]

**Comment by tiberiuichim** [comment]
Can we separate all this SEO and content image functionality in a new component? Ideally even one that's plugged, so that it can be overriden. I feel that View.jsx should act just as a "switch" and decide which view component to actually use. If we need something that should be rendered along with everything else, fine, but let's split that into another component, as it's a "separate concern".

**Review by tiberiuichim — CHANGES_REQUESTED** [review]
Can we separate all this SEO and content image functionality in a new component? Ideally even one that's plugged, so that it can be overriden. I feel that View.jsx should act just as a "switch" and decide which view component to actually use. If we need something that should be rendered along with everything else, fine, but let's split that into another component, as it's a "separate concern".

**Comment by pnicolli** [comment]
> Can we separate all this SEO and content image functionality in a new component? Ideally even one that's plugged, so that it can be overriden. I feel that View.jsx should act just as a "switch" and decide which view component to actually use. If we need something that should be rendered along with everything else, fine, but let's split that into another component, as it's a "separate concern".

@tiberiuichim 
If I understand correctly, you mean two things:

- refactor ``getContentImageInfo`` in a separate helper function
- create a separate component for all the content meta tags

Is this right? Any suggestions about where you would put the two new items in the source tree?

**Comment by tiberiuichim** [comment]
I'd put all of this into a ContentMetadataTags component. @sneridagh what's your take on this?

```
          {this.props.content.language && (
            <html lang={this.props.content.language.token} />
          )}
          <title>
            {this.props.content.seo_title || this.props.content.title}
          </title>
          <meta
            name="description"
            content={
              this.props.content.seo_description ||
              this.props.content.description
            }
          />
          <meta
            property="og:title"
            content={
              this.props.content.opengraph_title ||
              this.props.content.seo_title ||
              this.props.content.title
            }
          />
          <meta
            property="og:url"
            content={
              this.props.content.seo_canonical_url ||
              toPublicURL(this.props.content['@id'])
            }
          />
          {contentImageInfo.contentHasImage && (
            <meta
              property="og:image"
              content={toPublicURL(contentImageInfo.url)}
            />
          )}
          {contentImageInfo.contentHasImage && (
            <meta property="og:image:width" content={contentImageInfo.width} />
          )}
          {contentImageInfo.contentHasImage && (
            <meta
              property="og:image:height"
              content={contentImageInfo.height}
            />
          )}
          {(this.props.content.opengraph_description ||
            this.props.content.seo_description ||
            this.props.content.description) && (
            <meta
              property="og:description"
              content={
                this.props.content.opengraph_description ||
                this.props.content.seo_description ||
                this.props.content.description
              }
            />
          )}
          <meta name="twitter:card" content="summary_large_image" />
```

**Comment by sneridagh** [comment]
I agree, it's more sane and flexible. Once done also, as you suggested, it could also be wrappped into a pluggable.

**Comment by sneridagh** [comment]
I'll take care

**Comment by sneridagh** [comment]
@nzambello @giuliaghisini @pnicolli @tiberiuichim I double checked the add-ons and I think 
that it was the other way around, so the `opengraph_image` wins over the set `image`. Please could you confirm this?

Also, I added a new setting for specifying the image field to use, just in case.

@tiberiuichim moved to its own component, put a Pluggable in place `content-metadata-tags`. Also, time to open a Default Pluggables present in Volto documentation section?

File: docs/source/configuration/settings-reference.md [hunk 3]
```diff
@@ -246,3 +246,9 @@ in the `config.settings.serverConfig` object.
     For the moment it admits only one property: `errorPages` whose value is a Boolean.
 
     If `extractScripts.errorPages` is `true`, the JS will be inserted into the error page.
+
+### contentMetadataTagsImageField
+
+!!! block ""
+
+    The SEO default image field name for using it in the metadata HEAD tag. Defaults to `image`.
```
Comment (tiberiuichim) [inline docs/source/configuration/settings-reference.md:254]: > The OpenGraph image that will represent this content item, will be used in the metadata HEAD tag as `og:image` for SEO purposes. Defaults to `image`. See the [OpenGraph Protocol](https://ogp.me/) for more details.

**Review by tiberiuichim — APPROVED** [review]
LGTM, I've added a doc suggestion. @sneridagh I think you'll also need to add yourself to the changelog

**Comment by nzambello** [comment]
@sneridagh Thank you for taking this up, I'd be late on this.
I cannot approve my own PR, but I approve the chages 👍

**Pull request merged**